Sming/Arch/Host/app.mk

```diff
@@ -12,6 +12,11 @@ LDFLAGS += \
 # Executable
 TARGET_OUT_0			:= $(FW_BASE)/$(APP_NAME)$(TOOL_EXT)
 
+# Hosted Settings
+ifneq ($(ENABLE_HOSTED),)
+	COMPONENTS_AR := $(USER_LIBDIR)/$(CLIB_PREFIX)Hosted-$(CMP_Hosted_LIBHASH).a $(USER_LIBDIR)/$(CLIB_PREFIX)Hosted-Lib-$(CMP_Hosted-Lib_LIBHASH).a $(COMPONENTS_AR)
```

Comment:
@mikee47 Is it possible with the existing build system for a component to put its library at the front of the list of compiled libraries?

Reply:
This is probably the only way. I'm guessing the reason for this is to get the linkage with weak functions working properly? Note that weak functions won't work at all in Windows! Function wrappers are more reliable - the `malloc_count` Component uses these a lot so a good reference.

Reply:
> Function wrappers are more reliable - the malloc_count Component uses these a lot so a good reference.

Ok, thanks for pointing this out.  The `host_init` weak function is replaced with wrapper.